language-design.md

```diff
@@ -16,19 +16,78 @@ are:
+problems, or find the "right" answer, so they are seen as having less creativity.
+In this way, C and C++ programmers are considered objective in their search for the solution,
+while web development and UI are seen as much more subjective. This gendered differentiation
+hurts everyone who does not conform to these stereotypes.
+
+> "PL/I has not grown much. It is, for the most part, just as it was when it first came
+> out. It may be that this is just from lack of use. The flip side is that the lack of use may
+> have two causes. Number one: PL/I was not designed to grow—it was designed to be all
+> things to all who program right from the start. Number two: for its time, it started out
+> large. No one knew all of PL/I; some said that no one could know all of PL/I"
+[Steele, 1998].
+
+This quote contrasts with the ideas that we discussed in class, specifically that languages cycle
+through developing into a DSL, then into a general purpose language, and then back and through
+the cycle. PL/I was designed to not grow, which is a different philosophy from the other
+languages we have encountered. It would be interesting to discover what drove the design 
+philosophy for PL/I because it is such a large, immutable language.
+
+> "A language design can no longer be a thing. It must be a pattern—a pattern for growth—a 
+> pattern for growing the pattern for defining the patterns that programmers can use for 
+> their real work and their main goal"
+[Steele, 1998].
+
+By thinking of language as a pattern, Steele turns the focus from the programmer to the user.
+By understanding the pattern, users can expand the language, which Steele calls "a pattern for
+growth." At the very least, users feel like they are contributing to an ongoing project as
+opposed to using the results of a finished project. As a bonus, user feedback and suggestions 
+can facilitate a language's growth. The difficulty for programmers is not being able to consider
+what they would like to see, but considering what a user of their language would like to see.
 
 ---
-
 **Question**
 
 How would you know a well-designed language? What are the symptoms? How would
 you know a poorly designed language? What are the symptoms?
 
 **Response**
-
-
+A well-designed language can grow into your project. If you have an idea, a well-designed 
+language will have a way to help you achieve that vision, while a 
+poorly-designed language forces you to significantly change your idea to adapt to what the
+language can accomplish. One thing we disagree on is whether a well-designed language must 
+always have a large, helpful community. Sam thinks that a community focused on solving user problems
+can help languages grow since they should please the users. But Tiffany thinks that a large community only 
+reflects how widespread a language is. For example, JavaScript has a large community, but 
+many people claim it is poorly designed. 
```

Comment:
I would say that this is true, but at the same time JavaScript has been developing a lot, in no small part due to complaints by the large community, and new versions of ECMAScript (ES6 and 7) are quite pleasant to work with. You're both right :smiley: